extensions/panache/mongodb-panache-kotlin/deployment/src/test/kotlin/io/quarkus/mongodb/panache/kotlin/deployment/TestEnhancers.java

```diff
@@ -0,0 +1,192 @@
+        evaluate(new ByteCodeType(StudentRepository.class), new ByteCodeType(Student.class), PanacheMongoRepositoryBase.class, LONG);
+    }
+
+    /**
+     * we don't generate labels, e.g., like the compiler does (it's sourcemap related) so we need to elide some lines
+     * to account for the difference.
+     *
+     * @return
+     */
+    private List<Object> cleanUp(List<Object> input) {
+        List<Object> results = new ArrayList<>(input);
+        ListIterator<Object> iterator = results.listIterator();
+        while (iterator.hasNext()) {
+            Object next = iterator.next();
+            if (next instanceof String) {
+                String line = ((String) next).trim();
+                if (line.startsWith("LINENUMBER") || LABEL.matcher(line).matches()) {
+                    iterator.remove();
+                } else if (line.startsWith("LOCALVARIABLE")) {
+                    iterator.set(line.replaceAll(" L\\d.*", "\n"));
+                } else if (line.startsWith("IFNONNULL") || line.startsWith("LDC") && line.contains("\\u2026")) {
+                    iterator.set(line.substring(0, line.indexOf(" ") + 1) + "\n");
+                }
+            }
+        }
+        return results;
+    }
+
+    private void evaluate(ByteCodeType testClass, ByteCodeType entityType, Class<?> baseType, ByteCodeType idType) throws Exception {
+        ByteCodeType object = new ByteCodeType(Object.class);
+        Map<String, BytecodeMethod> localMethods = findLocalMethods(testClass);
+
+        Map<String, MethodNode> bridgeMethods = getBridgeMethods(baseType);
+        List<String> missing = bridgeMethods.keySet().stream()
+                .map(key -> {
+                    if (key.contains("ById")) {
+                        key = key.replace(object.descriptor(), idType.descriptor());
+                        if (key.contains("findById")) {
+                            key = key.substring(0, key.lastIndexOf(')') + 1) + entityType.descriptor();
+                        }
+                    }
+                    return "target_" + key;
+                })
+                .filter(key -> localMethods.get(key) == null)
+                .map(key -> String.format("%nCouldn't find %s in %s", key, testClass.dotName()))
+                .collect(Collectors.toList());
+        assertTrue(missing.isEmpty(), missing.toString());
+
+        localMethods.entrySet().stream()
+                .filter(e -> e.getKey().startsWith("target_"))
+                .forEach(entry -> {
+                    String key = entry.getKey().substring(7);
+                    BytecodeMethod generatedMethod = localMethods.get(key);
+                    BytecodeMethod targetMethod = entry.getValue();
+                    try {
+                        validate(key, generatedMethod.body, targetMethod.body);
+                    } catch (NullPointerException e) {
+                        throw new RuntimeException("could not find " + key);
+                    }
```

Comment:
Never try/catch NPE ! OK it's not that important but it's really an anti-pattern.

Reply:
Anyway, it's a test, so let's just make it fails with an exception

Reply:
it does fail with an exception.  this catch just ensures that the missing key (read: method) is logged so the dev knows where to start looking.  Otherwise the error is off in the middle of random code with little context as to the missing method.

But catching NPE is perfectly for reasons as stated here and others.